src/main/java/uk/gov/hmcts/reform/cpo/config/SecurityConfiguration.java

```diff
@@ -0,0 +1,99 @@
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationFilter;
+import uk.gov.hmcts.reform.authorisation.filters.ServiceAuthFilter;
+import uk.gov.hmcts.reform.cpo.security.JwtGrantedAuthoritiesConverter;
+
+import static org.springframework.security.config.http.SessionCreationPolicy.STATELESS;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    @Value("${spring.security.oauth2.client.provider.oidc.issuer-uri}")
+    private String issuerUri;
+
+    @Value("${oidc.issuer}")
+    private String issuerOverride;
+
+    private final ServiceAuthFilter serviceAuthFilter;
+    private final JwtAuthenticationConverter jwtAuthenticationConverter;
+
+    private static final String[] AUTH_ALLOWED_LIST = {
+        "/swagger-resources/**",
+        "/swagger-ui/**",
+        "/webjars/**",
+        "/v2/api-docs",
+        "/health",
+        "/health/liveness",
+        "/health/readiness",
+        "/info",
+        "/favicon.ico",
+        "/"
+    };
+
+    @Autowired
+    public SecurityConfiguration(final ServiceAuthFilter serviceAuthFilter,
+                                 final JwtGrantedAuthoritiesConverter jwtGrantedAuthoritiesConverter) {
+        super();
+        this.serviceAuthFilter = serviceAuthFilter;
+        jwtAuthenticationConverter = new JwtAuthenticationConverter();
+        jwtAuthenticationConverter.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthoritiesConverter);
+    }
+
+    @Override
+    public void configure(WebSecurity web) {
+        web.ignoring().antMatchers(AUTH_ALLOWED_LIST);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .addFilterBefore(serviceAuthFilter, BearerTokenAuthenticationFilter.class)
+            .sessionManagement().sessionCreationPolicy(STATELESS).and()
+            .csrf().disable()
+            .formLogin().disable()
+            .logout().disable()
+            .authorizeRequests()
+            .anyRequest()
+            .authenticated()
+            .and()
+            .oauth2ResourceServer()
+            .jwt()
+            .jwtAuthenticationConverter(jwtAuthenticationConverter)
+            .and()
+            .and()
+            .oauth2Client();
+    }
+
+    @Bean
+    @SuppressWarnings("PMD")
+    JwtDecoder jwtDecoder() {
+        NimbusJwtDecoder jwtDecoder = (NimbusJwtDecoder) JwtDecoders.fromOidcIssuerLocation(issuerUri);
+        // We are using issuerOverride instead of issuerUri as SIDAM has the wrong issuer at the moment
+        OAuth2TokenValidator<Jwt> withTimestamp = new JwtTimestampValidator();
+        OAuth2TokenValidator<Jwt> withIssuer = new JwtIssuerValidator(issuerOverride);
+        // FIXME : enable `withIssuer` once idam migration is done RDM-8094
```

Comment:
Recently in one of the project sonar is complaining about this block. You can remove this FIXME and commented lines altogether. Idam is not going to enforce that soon.

Reply:
👍